KeywordSearch/src/org/sleuthkit/autopsy/keywordsearch/LanguageDetector.java

```diff
@@ -0,0 +1,55 @@
+package org.sleuthkit.autopsy.keywordsearch;
+
+import com.optimaize.langdetect.LanguageDetectorBuilder;
+import com.optimaize.langdetect.i18n.LdLocale;
+import com.optimaize.langdetect.ngram.NgramExtractors;
+import com.optimaize.langdetect.profiles.LanguageProfile;
+import com.optimaize.langdetect.profiles.LanguageProfileReader;
+import com.optimaize.langdetect.text.CommonTextObjectFactories;
+import com.optimaize.langdetect.text.TextObject;
+import com.optimaize.langdetect.text.TextObjectFactory;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+class LanguageDetector {
+
+  public enum Language {
+    JAPANESE,
+    ENGLISH,
+  }
+
+  private List<LanguageProfile> languageProfiles;
+
+  LanguageDetector() {
+    try {
+      languageProfiles = Arrays.asList(
+          new LanguageProfileReader().readBuiltIn(LdLocale.fromString("en")),
+          new LanguageProfileReader().readBuiltIn(LdLocale.fromString("ja"))
+      );
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  Optional<Language> detect(String text) {
+    com.optimaize.langdetect.LanguageDetector languageDetector = LanguageDetectorBuilder.create(NgramExtractors.standard())
+        .withProfiles(languageProfiles)
+        .build();
+
+    TextObjectFactory textObjectFactory = CommonTextObjectFactories.forDetectingOnLargeText();
+
+    TextObject textObject = textObjectFactory.forText(text);
+    return languageDetector.detect(textObject).transform(Optional::of).or(Optional.empty()).map(LdLocale::getLanguage).flatMap(LanguageDetector::toLanguage);
+  }
+
+  private static Optional<Language> toLanguage(String s) {
```

Comment:
Perhaps this can be replaced by something else in the consolidated enum I suggest above. There are a number of different ways you could do this, e.g. you might leverage Enum.valueOf().